python/gen3authz/client/arborist/client.py

```diff
@@ -532,12 +532,31 @@ def list_policies(self):
         """
         return self.get(self._policy_url).json
 
-    def update_policy(self, policy_id, policy_json):
+    def update_policy(self, policy_id, policy_json, create_if_not_exist=False):
         """
         Arborist will create policy if not exist and overwrite if exist.
         """
-        url = self._policy_url + urllib.quote(policy_id)
-        response = self.put(url, json=policy_json)
+        if policy_json.get("id") and policy_json.get("id", "") != policy_id:
```

Comment:
We need to set `policy_json["id"] = policy_id` if `not policy_json.get("id")` right? It's only that we don't want the warning in this case.

Reply:
Not anymore, because Arborist will populate id from the URL anyway, and we made id an optional field in the unmarshal, per our discussion--right? All of this being because PUT policy doesn't care about the policy id in the json. 
Of course, we _can_ still do it, it's just that we don't _need_ to

Reply:
Ah my mind was stuck with the old logic 🤦‍♂yes you're right.